derive-encode/src/lib.rs

```diff
@@ -70,9 +70,87 @@ pub fn derive_encode(input: TokenStream) -> TokenStream {
             }
         }
     };
+
+    if cfg!(feature = "protobuf") {
+        let protobuf = derive_protobuf_encode(ast);
+        gen = quote! {
+            #gen
+
+            #protobuf
+        }
+    }
+
     gen.into()
 }
 
+fn derive_protobuf_encode(ast: DeriveInput) -> TokenStream2 {
+    let name = &ast.ident;
+
+    match ast.data {
+        syn::Data::Struct(s) => match s.fields {
+            syn::Fields::Named(syn::FieldsNamed { named, .. }) => {
+                let push_labels: TokenStream2 = named
+                    .into_iter()
+                    .map(|f| {
+                        let ident = f.ident.unwrap();
+                        let ident_string = KEYWORD_IDENTIFIERS
+                            .iter()
+                            .find(|pair| ident == pair.1)
+                            .map(|pair| pair.0.to_string())
+                            .unwrap_or_else(|| ident.to_string());
+
+                        quote! {
+                            let mut label = prometheus_client::encoding::proto::Label::default();
+                            label.name = #ident_string.to_string();
+                            label.value = format!("{}", self.#ident);
```

Comment:
This imposes a `Display` requirement on the fields of a struct. Is this actually needed? if so it would be good to document it somewhere because it's unexpected

Reply:
Thank you for pointing it out. I'm working on that in https://github.com/ackintosh/client_rust/pull/9.

Reply:
ackintosh#9 has been merged into this PR.